core/src/main/scala/org/apache/spark/sql/CachedDataFrame.scala

```diff
@@ -412,7 +412,6 @@ object CachedDataFrame
     val buffer = new Array[Byte](4 << 10) // 4K
     // final output is written to this buffer
     val output = new ByteBufferDataOutput(4 << 10,
-      DirectBufferAllocator.instance(),
```

Comment:
Now this will be outside of all accounting. Won't this result in a genuine OOME for large results? Please test with the original "select * from largeTable" case. I think better option will be to catch the DirectByteBuffer OOME here and convert to a LowMemoryException.